ko/pl.rst

```diff
@@ -0,0 +1,1383 @@
+            put_line(given_name || ' ' || family_name);     -- 문자열 병합 연산자 ||
+        END LOOP;
+    END;
+
+..
+    ******************
+    예외 처리
+    ******************
+..
+
+시스템 Exception
+======================
+
+PL/CUBSQL은 다른 많은 프로그래밍 언어와 마찬가지로 Exception 핸들러를 통한 에러 처리를 지원한다
+(참고: :ref:`Block 실행문 <block_stmt>`).
+사용자가 프로그램 선언부에서 자신만의 Exception을 정의할 수 있지만,
+주요 예외 상황에 대해서는 다음과 같이 시스템 Exception들이 미리 정의되어 있다.
+
++---------------------+------------------------------------------------------------------+
+| CASE_NOT_FOUND      | CASE 문에서 조건이 참인 WHEN 절이 없고 ELSE 절도 없음            |
++---------------------+------------------------------------------------------------------+
+| CURSOR_ALREADY_OPEN | 이미 열려 있는 커서에 다시 열기 시도                             |
++---------------------+------------------------------------------------------------------+
+| DUP_VAL_ON_INDEX    | Unique 조건이 걸려 있는 테이블 컬럼에 중복값 저장 시도           |
++---------------------+------------------------------------------------------------------+
+| INVALID_CURSOR      | 허용되지 않는 커서 조작 (예: 열려 있지 않은 커서를 닫으려고 함)  |
++---------------------+------------------------------------------------------------------+
+| LOGIN_DENIED        | 유효하지 않는 사용자 이름이나 암호로 DBMS에 로그인 시도          |
++---------------------+------------------------------------------------------------------+
+| NO_DATA_FOUND       | SELECT INTO 문 실행 결과 0개의 Row가 반환됨                      |
++---------------------+------------------------------------------------------------------+
+| PROGRAM_ERROR       | 시스템 내부 에러                                                 |
++---------------------+------------------------------------------------------------------+
+| ROWTYPE_MISMATCH    | 조회 결과와 결과를 담을 호스트 변수 사이에 타입이 호환되지 않음  |
++---------------------+------------------------------------------------------------------+
+| STORAGE_ERROR       | 메모리 부족으로 인한 할당 실패                                   |
++---------------------+------------------------------------------------------------------+
+| TOO_MANY_ROWS       | SELECT INTO 문 실행 결과 2개 이상의 Row가 반환됨                 |
++---------------------+------------------------------------------------------------------+
+| VALUE_ERROR         | 명시적/암묵적 형변환 실패                                        |
++---------------------+------------------------------------------------------------------+
+| ZERO_DIVIDE         | 0으로 나누기 시도                                                |
++---------------------+------------------------------------------------------------------+
+
+.. code-block:: sql
+
+    CREATE FUNCTION athlete_code(name VARCHAR) RETURN integer AS
+        c INTEGER;
+    BEGIN
+        -- SELECT INTO 문은 단 하나, 그리고 오직 하나의 Row를 결과로 가져야 함
+        SELECT code INTO c FROM athlete a WHERE a.name = name;
+        RETURN c;
+    EXCEPTION
+        WHEN NO_DATA_FOUND THEN
+            PUT_LINE('error: no rows found for athlete name ' || name);
+            RETURN -1;
```

Comment:
Are we going to support "RETURN NO_DATA_FOUND;" ?

Reply:
No. exception is not a value in PL/CUBSQL, which means it cannot be assigned to a variable, handed over to a routine as an argument, and be a return value.